src/Sentry/Internal/AotHelper.cs

```diff
@@ -11,18 +11,14 @@ private class AotTester
 
     internal static bool IsAot { get; }
 
+    [UnconditionalSuppressMessage("Trimming", "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code", Justification = AotHelper.SuppressionJustification)]
     static AotHelper()
     {
-        try
-        {
-            // GetMethod should throw an exception if Trimming is enabled
-            var type = typeof(AotTester);
-            _ = type.GetMethod(nameof(AotTester.Test));
-        }
-        catch
-        {
-            IsAot = true;
-        }
+#if NET6_0_OR_GREATER   // TODO NET7 once we target it
```

Comment:
We can keep this forever technically, since it'll include NET7 and in a way it helps document what's the min version this works from. Also avoid us having to go through all the code base and change stuff that won't do any change (I did that a bit on my last PR dropping stuff)

```suggestion
#if NET6_0_OR_GREATER
```